src/renderer.js

```diff
@@ -41,6 +41,58 @@ document.addEventListener("DOMContentLoaded", () => {
       ipcRenderer.send("new-window");
     });
 
+    nav.addEventListener("toggle-bookmark", async () => {
+      const urlInput = nav.querySelector("#url");
+      if (!urlInput || !urlInput.value) {
+        console.error("URL input is empty, cannot toggle bookmark.");
+        return;
+      }
+      const url = urlInput.value.trim();
+      
+      const bookmarks = await ipcRenderer.invoke("get-bookmarks");
+      const existingBookmark = bookmarks.find((b) => b.url === url);
+
+      if (existingBookmark) {
+        ipcRenderer.invoke("delete-bookmark", { url });
+      } else {
+        const title = pageTitle.innerText
+          .replace(" - Peersky Browser", "")
+          .trim();
+        let favicon = "";
+        try {
+          const iconLink = document.querySelector(
+            'link[rel="icon"], link[rel="shortcut icon"]'
+          );
+          if (iconLink) {
+            favicon = new URL(iconLink.href, url.origin).href;
+          }
+
+          if (!favicon) {
+        
+            favicon = `https://www.google.com/s2/favicons?domain=${url.hostname}`;
+          }
+        } catch (e) {
+          console.error("Error fetching favicon:", e);
+          favicon = `https://www.google.com/s2/favicons?domain=${url.hostname}`;
```

Comment:
Relying on the external Google favicon service as a final fallback means favicon loading will fail when offline or in restricted network environments.

Let's have it locally and load it from `peersky://static/assets/svg/`

Reply:
If we can download that google default favicon as SVG, then it would be great.
Else we can use https://icons.getbootstrap.com/?q=glob